metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1142,9 +1190,20 @@ public ControllerResult<Void> generateRecordsAndResult() throws Exception {
                         "at metadata.version {} from {}.", bootstrapMetadata.records().size(),
                         bootstrapMetadata.metadataVersion(), bootstrapMetadata.source());
                 records.addAll(bootstrapMetadata.records());
-            } else if (featureControl.metadataVersion().equals(MetadataVersion.MINIMUM_KRAFT_VERSION)) {
-                log.info("No metadata.version feature level record was found in the log. " +
+
+                // Bootstrap the initial ZK Migration record
+                featureControl.generateZkMigrationRecord(
```

Comment:
I think we can just directly create the record here if the metadata version supports it. If the MV does not support it but the static configuration is set, we should throw a non-ApiException exception which will trigger a fault handler